### PR TITLE
added etcdctl options to resource

### PR DESCRIPTION
**Pull request by willejs**
I have added an option to add additional options to the etcdctl command when configuring flannel.

### CODE REVIEW

**Comment by willejs** [comment]
@aespinosa can you review please?

File: libraries/command_generators.rb [hunk 1]
```diff
@@ -77,8 +78,16 @@ def etcdctl_bin
...
+      if etcdctl_options
+        " #{etcdctl_options}"
+      else
+        ""
+      end
+    end
+
     def etcdctl_command
-      "#{etcdctl_bin} set #{etcd_prefix} '#{configuration.to_json}'"
+      "#{etcdctl_bin}#{etcdctl_opts} set #{etcd_prefix} '#{configuration.to_json}'"
```
Comment (aespinosa) [inline libraries/command_generators.rb]: does having two spaces when `etcdctl_options` is nil break the configuration?  can lines 81-88 be removed?

Comment (willejs) [inline libraries/command_generators.rb]: @aespinosa It probably wont break it, if you prefer that i could do it, i thought this was more elegant, but really its just more code?

Comment (aespinosa) [inline libraries/command_generators.rb]: yes please. i prefer less code for now.

**Comment by aespinosa** [comment]
Added tests and addressed my comments

**Pull request merged**